.github/workflows/pull_requests.yml

```diff
@@ -93,3 +93,64 @@ jobs:
             sudo mv /tmp/yq /usr/local/bin/yq
       - name: test
         run: make test
+
+  ##############################################################################
+  # Integration tests
+  ##############################################################################
+
+  setup-integration-tests:
+    runs-on: ubuntu-20.04
+    outputs:
+      kind_images: ${{ steps.set_kind_images.outputs.kind_images }}
+      test_names: ${{ steps.set_test_names.outputs.test_names }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: set_kind_images
+        name: Set kind images
+        working-directory: ./tests/integration/
+        run: |
+          echo "::set-output name=kind_images::$(jq -cM '.supported' kind_images.json)"
+      - name: Print kind images
+        run: echo "Kind images ${{ steps.set_kind_images.outputs.kind_images }}"
+      - id: set_test_names
+        name: Set test names
+        working-directory: ./tests/integration/
+        # grep magic described in https://unix.stackexchange.com/a/13472
+        run: |
+          echo "::set-output name=test_names::$(grep -rhoP "(?<=^func )(Test[a-zA-z_0-9]+)(?=\(t \*testing.T\) {)" . | jq -R . | jq -cs .)"
```

Comment:
Is there really no cleaner way to do this?

Reply:
There might be but I don't know about it 🤷 

`go test -list` returns includes the output of the e.g. what `init()` prints out:

```
go test -v -list Test_Helm
2021/11/29 14:38:51 Successfully read kind images spec
2021/11/29 14:38:51 Default kind image: kindest/node:v1.22.0@sha256:b8bda84bb3a190e6e028b1760d277454a72267a5454b57db34437c34a588d047
2021/11/29 14:38:51 Supported kind images: [kindest/node:v1.19.11@sha256:07db187ae84b4b7de440a73886f008cf903fcf5764ba8106a9fd5243d6f32729 kindest/node:v1.20.7@sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9 kindest/node:v1.21.2@sha256:9d07ff05e4afefbba983fac311807b3c17a5f36e7061f6cb7e2ba756255b2be4 kindest/node:v1.22.0@sha256:b8bda84bb3a190e6e028b1760d277454a72267a5454b57db34437c34a588d047]
Test_Helm_Default
ok      github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration  0.483s
```

Reply:
Using JSON doesn't help much:

```
go test -v -list Test_Helm -json
{"Time":"2021-11-29T14:39:54.949815+01:00","Action":"output","Package":"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration","Output":"2021/11/29 14:39:54 Successfully read kind images spec\n"}
{"Time":"2021-11-29T14:39:54.950129+01:00","Action":"output","Package":"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration","Output":"2021/11/29 14:39:54 Default kind image: kindest/node:v1.22.0@sha256:b8bda84bb3a190e6e028b1760d277454a72267a5454b57db34437c34a588d047\n"}
{"Time":"2021-11-29T14:39:54.950139+01:00","Action":"output","Package":"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration","Output":"2021/11/29 14:39:54 Supported kind images: [kindest/node:v1.19.11@sha256:07db187ae84b4b7de440a73886f008cf903fcf5764ba8106a9fd5243d6f32729 kindest/node:v1.20.7@sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9 kindest/node:v1.21.2@sha256:9d07ff05e4afefbba983fac311807b3c17a5f36e7061f6cb7e2ba756255b2be4 kindest/node:v1.22.0@sha256:b8bda84bb3a190e6e028b1760d277454a72267a5454b57db34437c34a588d047]\n"}
{"Time":"2021-11-29T14:39:54.950275+01:00","Action":"output","Package":"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration","Output":"Test_Helm_Default\n"}
{"Time":"2021-11-29T14:39:54.950538+01:00","Action":"output","Package":"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration","Output":"ok  \tgithub.com/SumoLogic/sumologic-kubernetes-collection/tests/integration\t0.335s\n"}
{"Time":"2021-11-29T14:39:54.950558+01:00","Action":"pass","Package":"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration","Elapsed":0.335}
```

Reply:
Allright, I guess that's what it'll have to be. Hopefully this doesn't bite us. If it does. we can always consider keeping the test list in a separate file and generating it manually when adding more tests.